tests/integrationv2/test_signature_algorithms.py

```diff
@@ -0,0 +1,176 @@
+import copy
+import pytest
+
+from configuration import available_ports, ALL_TEST_CIPHERS, ALL_TEST_CURVES, ALL_TEST_CERTS, PROVIDERS, PROTOCOLS
+from common import ProviderOptions, Protocols, Ciphers, Certificates, Signatures, data_bytes
+from fixtures import managed_process
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name, get_expected_s2n_version
+
+
+certs = [
+    Certificates.RSA_2048_SHA256,
+    Certificates.RSA_2048_SHA384,
+    Certificates.RSA_PSS_2048_SHA256,
+    Certificates.ECDSA_256,
+    Certificates.ECDSA_384,
+]
+
+all_sigs = [
+    Signatures.RSA_SHA1,
+    Signatures.RSA_SHA224,
+    Signatures.RSA_SHA256,
+    Signatures.RSA_SHA384,
+    Signatures.RSA_SHA512,
+    Signatures.ECDSA_SECP256r1_SHA256,
+    Signatures.RSA_PSS_SHA256,
+]
+
+unsupported_ciphers = [
+    Ciphers.AES128_SHA,
+    Ciphers.AES256_SHA,
+    Ciphers.AES128_SHA256,
+    Ciphers.AES256_SHA256,
+    Ciphers.AES128_GCM_SHA256,
+    Ciphers.AES256_GCM_SHA384,
+]
+
+
+def skip_ciphers(*args, **kwargs):
+    cert = kwargs.get('certificate')
+    cipher = kwargs.get('cipher')
+    protocol = kwargs.get('protocol')
+    sigalg = kwargs.get('signature')
+
+    if not cert.compatible_with_cipher(cipher):
+        return True
+
+    if not cert.compatible_with_sigalg(sigalg):
+        return True
+
+    if protocol is Protocols.TLS13 and sigalg.min_protocol is not Protocols.TLS13:
+        return True
+
+    if protocol < sigalg.min_protocol:
+        return True
+
+    if cipher in unsupported_ciphers:
+        return True
+
+    return invalid_test_parameters(*args, **kwargs)
+
+
+@pytest.mark.uncollect_if(func=skip_ciphers)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("protocol", [Protocols.TLS13, Protocols.TLS12], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("signature", all_sigs, ids=get_parameter_name)
+@pytest.mark.parametrize("client_auth", [True, False], ids=get_parameter_name)
+def test_s2n_server_signature_algorithms(managed_process, cipher, provider, protocol, certificate, signature, client_auth):
+    port = next(available_ports)
+
+    # Two unsupported signatures to verify unsupported sigs are ignored
+    sigs = ["ECDSA+SHA256", "ECDSA+SHA512", signature.name]
```

Comment:
nit: Could you be more specific  in the comment, on why these two signatures are unsupported signatures?

Reply:
This is copy / pasted from the original test. I don't see any reasoning, but I didn't want to remove something the original test supported.

Reply:
I've gone ahead and removed this all together. I'm not sure exactly what it was testing, so if the original author can show that, we can re-implement it.